tests/FsCheck.Test/Gen.fs

```diff
@@ -119,6 +119,13 @@ module Gen =
         Gen.listOfLength length (Gen.constant v)
         |> sample1
         |> ((=) (List.init length (fun _ -> v)))
+
+    [<Property>]
+    let Shuffle (xs:Set<int>) =
+        not xs.IsEmpty ==> lazy
+        Gen.shuffle xs
+        |> sample 10
+        |> List.exists ((<>) (Set.toSeq xs))
```

Comment:
This test can be strengthened I think: taking an input list (see `NonEmptyList` and `NonEmptySet`, by the way, to get rid of the precondition) and checking that all the elements in the shuffle appear in the original and vice versa?